ui/app/pages/swaps/index.js

```diff
@@ -231,6 +236,10 @@ export default function Swap() {
     return <Redirect to={{ pathname: DEFAULT_ROUTE }} />;
   }
 
+  if (!isFeatureFlagLoaded) {
```

Comment:
It looks like this loading screen is only shown once, the first time the user clicks 'swaps'. On subsequent navigations to the swaps page, it will show the maintenance page or the build quote page immediately while it checks the API liveness in the background.

Was this intended? At the very least it's probably worth showing the loading screen if the last check showed the API as down.

Reply:
I had the same thought as you at first, but we actually clean up a Swaps state by calling the `prepareToLeaveSwaps` function from various places, e.g. here: https://github.com/MetaMask/metamask-extension/blob/develop/ui/app/pages/swaps/index.js#L178
`prepareToLeaveSwaps` calls the `resetBackgroundSwapsState` function and that one calls `resetSwapsState` every time a user leaves the Swaps feature. That will set the `isFeatureFlagLoaded` flag to false.

I've just tried this by going back by 3 different ways: clicking on Cancel, clicking on the Fox head and clicking on the "Close" button on the "Offline for maintenance" page and then every time I click on "Swaps" from the main page, I can see the page loader again.

Reply:
Huh, I hadn't realized. Good to know!